Assets/Scripts/Game Scene/Obstacle/Obstacle.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using UnityEngine;
+
+public class Obstacle : MonoBehaviour
+{
+    [SerializeField] List<float> heightValues = new List<float>();
+    [SerializeField] float baseSpeed;
+
+    Action<Obstacle> releaseCallback;
+    readonly System.Random random = new(); //idk why VSC forced me to use System.Random instead of UnityEngine.Random
```

Comment:
UnityEngine.Random is a value to keep in memory, it's a utility class. You don't need to save that, you can keep use as:

```cs
float randomHeight = heightValues[UnityEngine.Random.Range(0, heightValues.Count)];
```

Also, Unity  Random is faster than System.Random

Reply:
I was using Unity Random yesterday and today morning, but a little before I commit the code, the VSC accused an error in this line about ambiguity and made me use System.Random 🤔 I'll change it.